receiver/dockerstatsreceiver/receiver.go

```diff
@@ -111,12 +116,264 @@ func (r *receiver) scrape(ctx context.Context) (pmetric.Metrics, error) {
 	md := pmetric.NewMetrics()
 	for res := range results {
 		if res.err != nil {
-			// Don't know the number of failed metrics, but one container fetch is a partial error.
+			// Don't know the number of failed stats, but one container fetch is a partial error.
 			errs = multierr.Append(errs, scrapererror.NewPartialScrapeError(res.err, 0))
 			continue
 		}
-		res.md.ResourceMetrics().CopyTo(md.ResourceMetrics())
+		res.md.ResourceMetrics().MoveAndAppendTo(md.ResourceMetrics())
```

Comment:
This is the bug I found regarding overwriting resource metrics. Only the last container stats was kept.